tests/vmi_lifecycle_test.go

```diff
@@ -93,6 +93,25 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 		tests.EnsureKVMPresent()
 	})
 
+	Context("when virt-launcher is deleted", func() {
+		It("should label the node with kubevirt.io/schedulable=false", func() {
+			pods, err := virtClient.CoreV1().Pods("").List(metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", v1.AppLabel, "virt-handler"),
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods.Items)).Should(BeNumerically(">", 0))
+
+			pod := pods.Items[0]
+			handlerNamespace := pod.GetNamespace()
+			err = virtClient.CoreV1().Pods(handlerNamespace).Delete(pod.Name, &metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			n, err := virtClient.CoreV1().Nodes().Get(pod.Spec.NodeName, metav1.GetOptions{})
```

Comment:
Just to be on safe side. I would use `Eventually` here.

Reply:
I had originally thought the same thing, but the delete call shouldn't return until the go routine is ran and exits. By this point it should have been set, if it is not set the update failed. If we check using `Eventually` that would mean that is was marked during pod start-up.

Reply:
I think @xpivarc is right. You actually have to use an Eventually function here. I wonder why it works at all. Setting the DeletionTimestamp on a Pod is completely independent from the kubelet which will send the SIGTERM.

Have a look here on how to ensure that virt-handler pod gets evicted just from the target node and is not rescheduled until you are done with your check: https://github.com/kubevirt/kubevirt/blob/master/tests/vmi_lifecycle_test.go#L609.

Reply:
Gottcha, makes sense. I have updated this to use `Eventually`